library/ssl_tls13_client.c

```diff
@@ -2413,13 +2573,19 @@ static int ssl_tls13_process_server_finished( mbedtls_ssl_context *ssl )
         return( ret );
     }
 
+#if defined(MBEDTLS_SSL_EARLY_DATA)
+    if( ssl->early_data_status == MBEDTLS_SSL_EARLY_DATA_STATUS_ACCEPTED )
+        mbedtls_ssl_handshake_set_state( ssl, MBEDTLS_SSL_END_OF_EARLY_DATA );
+    else
+#endif /* MBEDTLS_SSL_EARLY_DATA */
+    {
 #if defined(MBEDTLS_SSL_TLS1_3_COMPATIBILITY_MODE)
-    mbedtls_ssl_handshake_set_state(
-        ssl,
-        MBEDTLS_SSL_CLIENT_CCS_AFTER_SERVER_FINISHED );
+        mbedtls_ssl_handshake_set_state(
```

Comment:
We should not go into that state if we have proposed early data (but were rejected) as encryption is in place for outgoing data and we would send an encrypted CCS which is not allowed.